include/battle.h

```diff
@@ -371,7 +371,7 @@ struct AiLogicData
     bool8 weatherHasEffect; // The same as WEATHER_HAS_EFFECT. Stored here, so it's called only once.
     u8 mostSuitableMonId[MAX_BATTLERS_COUNT]; // Stores result of GetMostSuitableMonToSwitchInto, which decides which generic mon the AI would switch into if they decide to switch. This can be overruled by specific mons found in ShouldSwitch; the final resulting mon is stored in AI_monToSwitchIntoId.
     struct SwitchinCandidate switchinCandidate; // Struct used for deciding which mon to switch to in battle_ai_switch_items.c
-    u8 shouldSwitch[MAX_BATTLERS_COUNT]; // Stores result of ShouldSwitch, which decides whether a mon should be switched out
+    bool8 shouldSwitch; // Stores result of ShouldSwitch, which decides whether a mon should be switched out
```

Comment:
I'm not sure if you would consider this a bool because you have 5 different outcomes (4 battlers + nothing). I would keep the u4

Reply:
Oh duh that makes sense, went one step too far mirroring the pattern from the other bitfield, addressed :)

Reply:
would probably good to change that as well